### PR TITLE
Fix CoreAnimation related warnings (issue #833)

### PR DESCRIPTION
**Pull request by barijaona**
Fix issue #833 (and maybe #724)

### CODE REVIEW

**Comment by josh64x2** [comment]
@Eitot you're probably best to review this one 😅

File: src/RefreshManager.m [hunk 1]
```diff
@@ -33,6 +33,7 @@
 #import "NSNotificationAdditions.h"
 #import "VTPG_Common.h"
 #import "FeedItem.h"
+#import <QuartzCore/QuartzCore.h>
```
Comment (josh64x2) [inline src/RefreshManager.m]: I don't think we should be adding new UI to RefreshManager (and GoogleReader). This functionality would be better moved into another class.

Comment (barijaona) [inline src/RefreshManager.m]: I am not exactly adding UI, I am trying to fix the current one.
I noticed like @Eitot that problem seems [related to existing calls to ActivityItem](https://github.com/ViennaRSS/vienna-rss/issues/833#issuecomment-287895810), but from my tests, his suggestion was not sufficient and I needed the trick [found here](http://stackoverflow.com/questions/12297787/coreanimation-warning-deleted-thread-with-uncommitted-catransaction).

Comment (barijaona) [inline src/RefreshManager.m]: My [issue-833bis branch](https://github.com/barijaona/vienna-rss/tree/issue-833bis) might be slightly more elegant than this one. I'll close this one and submit a new pull request.

**Comment by barijaona** [comment]
Seems more reliable than #865

**Comment by josh64x2** [comment]
Curiously, I can't even reproduce #833 in current `master` as my Activity Window seems to be permanently greyed out and never has any entries appear.

**Comment by barijaona** [comment]
What I do for reproducing #833 : 
- After Vienna launch and an initial refresh of subscriptions,
- I relaunch a "Refresh All Subscriptions" with the toolbar
- afterwards I launch menu "Window -> Activity Window",
- and I fool around with the Activity Window, changing the sort criteria and order.

The problem mostly occurs with feeds returning 304 for HTTP answer code (no new articles since last refresh).

**Comment by josh64x2** [comment]
I can get Activity Window and the bug to show on `legacy3.1.x` branch, but not on `master`. The Activity Window just stays blank unless I step through the code which makes me think the bindings/notifications are not being set up in the correct order after https://github.com/ViennaRSS/vienna-rss/pull/832 in some weird circumstances. I'm having a look now...

**Comment by barijaona** [comment]
> Le 17 avr. 2017 à 09:08, Josh <notifications@github.com> a écrit :
> 
> I can get Activity Window and the bug to show on legacy3.1.x branch, but not on master. The Activity Window just stays blank unless I step through the code
> 

Maybe because the `activityItemStatusUpdatedNotification` isn’t handled ?

**Comment by josh64x2** [comment]
good point - that could be it, although i would have thought you'd be experiencing issues as well?

**Comment by barijaona** [comment]
If : 
- I disable "Check new articles on startup" in Preferences,
- open Activity Window __before__ launching a "refresh all"

then Activity Window does not get populated.

**Comment by josh64x2** [comment]
ahhh that must be what I am doing wrong. thanks!

**Comment by barijaona** [comment]
Note that with `legacy3.1.x`, the Activity Window gets populated with the above mentioned procedure, so something was probably broken by #832.

**Comment by barijaona** [comment]
Regarding #833 per se, you can alternatively look at my [legacy-issue833](https://github.com/barijaona/vienna-rss/tree/legacy-issue833) branch.

**Comment by josh64x2** [comment]
I'll have time to finish looking at this over the next day :)

**Comment by josh64x2** [comment]
I'm still not comfortable with `QuartzCore` being required in `RefreshManager` and `GoogleReader`. We need to improve on our separation of concerns. 

I'm going to tag this as **do not merge** for now, and I plan to try an alternative approach on Tuesday as there is a holiday in Australia. Unfortunately I didn't have as much free time this weekend as I would have liked.

**Comment by barijaona** [comment]
I can understand your concern for `master`, but how about merging my [legacy-issue833](https://github.com/barijaona/vienna-rss/tree/legacy-issue833) branch into `legacy-3.1.x` so we can publish a 3.1.10 build ?

**Comment by josh64x2** [comment]
@barijaona yup go ahead and merge it into the legacy branch, I have tested that the solution works so that is good enough for me :)

**Comment by barijaona** [comment]
Hi @josh64x2. The 7-days window for reviewing has elapsed for a long time for this pull request.
Unless you have an alternative nearly finished, I will merge it because development needs to move on…

**Comment by josh64x2** [comment]
Hi @barijaona, sorry I am going to have to decline this Pull Request. By introducing QuartzCore to `RefreshManager` and `GoogleReader` classes you are increasing the amount of UI code in non-UI classes which further violates the MVC pattern and Single responsibility principle we are trying to align with.

This root cause of this issue is mainly caused by threading problems in `RefreshManager` which should be fixed as part of my work to remove `ASIHTTPRequest` from `RefreshManager`. In addition to this I will begin refactoring `ActivityItem` and the `ActivityPanelController` which should fix the Activity information not displaying correctly in some circumstances.

These Core Animation warnings are not a major problem for users and your patch was already included in 3.1.10 so I think it is fine to let it wait a few more days for my `RefreshManager` work to be done.

**Comment by barijaona** [comment]
Hi @josh64x2 , I am sorry, but I'd prefer you to stick to your previous words. From my point of view, and given the current situation, blocking this PR seems out of proportion.

You'll agree with me that the mix of refresh code and UI code was preexisting to this pull request. As I already wrote, __I am not adding any new UI, I am just fixing the current one__. You tried to have something smarter, I tried too, but obviously this needs more work than we predicted.

There are exactly 9 lines you don't like (the Core Animation related ones) while we have [an issue](https://github.com/ViennaRSS/vienna-rss/issues/724#issuecomment-296420631) which undermines Vienna's credibility for six months now. We can easily reverse these 9 lines whenever we find a better solution, but for the time being, we need to be pragmatic and distinguish between what could be done (Vienna NG) and what can be done to deliver a functional Vienna 3.2 in a predictable future.

A few weeks ago, you complained that the 7 days review before merging your own code was too long and was holding back development. It has been 14 days now, so we need to move on ; I have just added some TODO comments encouraging some refactoring.

**Comment by josh64x2** [comment]
@barijaona I did review this change within 7 days and I made it clear I wasn't happy with the approach, effectively completing the review but not approving the PR. If @Eitot is happy to approve it then you may merge.

There is some pre-existing UI code in the classes, but we need to aim to reduce the amount of UI code in non-UI classes, not increase it. 

By merging this PR we will be replacing a very minor bug with 0 current impact on users, with a bug-fix that increases our technical debt.

I disagree that it is undermining the credibility of Vienna as this issue was never user-facing, and isn't in any current public release of Vienna. v3.2 is not going to be released in the next week or two and so holding off will not accept

**Comment by josh64x2** [comment]
Please note: This is not personal - it is purely a technical reason. I would not accept such a PR at work either. 

However, my biggest concern for Vienna is that we might be appearing fractious. If you feel very strongly about merging in this PR then you may and I will not stop you.

**Comment by barijaona** [comment]
I can attest that the issue is real because I am able with the current master to reproduce crashes.
There is not yet a compelling reason to release v3.2, but as an open source project, we'd better continue releasing beta versions. Working around a potential crash is a matter of courtesy toward our beta testers.

**Pull request merged**